### ISSUE TITLE
remove 2-way binding of current step?

### ISSUE THREAD

**Issue by lougreenwood** [alexlafroscia/ember-steps#135]
hello! 🙂

I wondered if you had any thoughts on what ember-steps might look like if the `currentStep` argument were to have 2-way binding removed, possibly something like:

```
<StepManager
  @currentStep={{this.currentStep}}
  @onStepChange={{action (mut this.currentStep}}
  as |W|
>
 ....
```

I'm currently trying to externally drive the current step with a computed property on my component, but it seems that as soon as I transition into some other step, then the computed property is overwritten because of 2-way binding.

**Comment by lougreenwood** [comment]
I think I may take a look at this, here's my plan, @alexlafroscia - how's does this sound to you?

* Add a new `onStepChange` arg which expects an action and is triggered when internal currentStep changes
* if  `@onStepChange` arg **is not** passed to StepManager, then use existing 2-way binding (`this.set('currentStep', currentStep)` or whatever)
* if `@onStepChange` arg **is** is passed to StepManager, disable 2-way binding and use action instead

Seems this would allow it to be a non breaking change and keep the door open for deprecating 2-way binding in future if there's an appetite for that.

**Comment by alexlafroscia** [comment]
Hey @lougreenwood! 

> but it seems that as soon as I transition into some other step, then the computed property is overwritten because of 2-way binding.

Can you explain a bit more about what you mean? Are you controlling the current step through both the computed property _and_ using the `transition-to` actions?

**Comment by lougreenwood** [comment]
> Are you controlling the current step through both the computed property _and_ using the `transition-to` actions?

Yep, I'm using the computed property to trigger various "primary" steps, the computed property is observing some Ember Data model from the store.

When on one of these "primary steps", the user may transition to some secondary step (outside the scope of the computed), this uses `transition-to`

The user may then trigger some operation when in one of these "secondary steps" - these operations are API calls which remotely save a record and update the store.

The computed property then observes changes on this model and resets to a specific step using `currentStep`.

**Comment by alexlafroscia** [comment]
I see -- the use-case makes sense to me for sure.

I'm not surprised that using both approaches doesn't actually work, and that the CP you define is blown away by the new value coming out of the step manager.

I'm a little hesitant to take the approach of making the `currentStep` property sometimes one-way bound and sometimes two-way bound. I think that could potentially be a foot-gun for people and the logic, internally, would be pretty complex.

I'm trying to think of there's any reason that we couldn't just make `currentStep` _only_ one-way bound with the setter method that you suggested. I don't think that would be a problem, and would help the addon adhere to DDAU. 

If you want to make that refactor, I would be happy to review a PR that changes that behavior!

**Comment by lougreenwood** [comment]
Cool, going all-in with DDAU sounds good - certainly will make for cleaner code!

I’ll try to take a look over the next few days. Thanks!

**Comment by lougreenwood** [comment]
closed by #146
